### PR TITLE
[voicerss] Hide the API key when logging API URL

### PR DESCRIPTION
**Pull request by lolodomo**
Also fix few typos

Also small changes regarding log levels and exception handling

Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

**Comment by lolodomo** [comment]
@jlaur @fwolter : please have a look to this very small PR, it concerns security.

PS: I will add null annotations in a next PR.

File: bundles/org.openhab.voice.voicerss/src/main/java/org/openhab/voice/voicerss/internal/cloudapi/CachedVoiceRSSCloudImpl.java [hunk 1]
```diff
@@ -75,12 +77,8 @@ public File getTextToSpeechAsFile(String apiKey, String text, String locale, Str
             writeText(txtFileInCache, text);
             // return from cache
             return audioFileInCache;
-        } catch (FileNotFoundException ex) {
-            logger.warn("Could not write {} to cache", audioFileInCache, ex);
-            return null;
         } catch (IOException ex) {
-            logger.error("Could not write {} to cache", audioFileInCache, ex);
-            return null;
+            throw new IOException("Could not write to cache file: " + ex.getMessage(), ex);
```
Comment (jlaur) [inline bundles/org.openhab.voice.voicerss/src/main/java/org/openhab/voice/voicerss/internal/cloudapi/CachedVoiceRSSCloudImpl.java:81]: Just checking: `IOException` is catched and handled somewhere, right? Previously it was not thrown from this method.

**Review by jlaur — COMMENTED** [review]
Just two questions for being sure.

File: bundles/org.openhab.voice.voicerss/src/main/java/org/openhab/voice/voicerss/internal/cloudapi/CachedVoiceRSSCloudImpl.java (hunk 1 above)
Comment (lolodomo) [reply bundles/org.openhab.voice.voicerss/src/main/java/org/openhab/voice/voicerss/internal/cloudapi/CachedVoiceRSSCloudImpl.java:81]: Yes, it is handled by the caller.
It was already thrown by this method, but not at this place (elsewhere in the method).

File: bundles/org.openhab.voice.voicerss/src/main/java/org/openhab/voice/voicerss/internal/VoiceRSSTTSService.java [hunk 2]
```diff
@@ -223,7 +220,7 @@ private String getApiAudioFormat(AudioFormat format) throws TTSException {
         }
     }
 
-    private CachedVoiceRSSCloudImpl initVoiceImplementation() {
+    private CachedVoiceRSSCloudImpl initVoiceImplementation() throws IllegalStateException {
```
[the remark this answers is not in the document]
Comment (lolodomo) [reply bundles/org.openhab.voice.voicerss/src/main/java/org/openhab/voice/voicerss/internal/VoiceRSSTTSService.java:223]: The exception is catched by the caller, the method `activate`
```
    protected void activate(Map<String, Object> config) {
        try {
            modified(config);
            voiceRssImpl = initVoiceImplementation();
            voices = initVoices();
            audioFormats = initAudioFormats();

            logger.debug("Using VoiceRSS cache folder {}", getCacheFolderName());
        } catch (IllegalStateException e) {
            logger.error("Failed to activate VoiceRSS: {}", e.getMessage(), e);
        }
```

File: bundles/org.openhab.voice.voicerss/src/main/java/org/openhab/voice/voicerss/internal/cloudapi/CachedVoiceRSSCloudImpl.java (hunk 1 above)
Comment (lolodomo) [reply bundles/org.openhab.voice.voicerss/src/main/java/org/openhab/voice/voicerss/internal/cloudapi/CachedVoiceRSSCloudImpl.java:81]: What I changed here is that the method will no more return null but will rather throw an exception.

File: bundles/org.openhab.voice.voicerss/src/main/java/org/openhab/voice/voicerss/internal/VoiceRSSTTSService.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (jlaur) [reply bundles/org.openhab.voice.voicerss/src/main/java/org/openhab/voice/voicerss/internal/VoiceRSSTTSService.java:223]: That's good, but any need to declare unchecked exception in method signature?

[the remark this answers is not in the document]
Comment (lolodomo) [reply bundles/org.openhab.voice.voicerss/src/main/java/org/openhab/voice/voicerss/internal/VoiceRSSTTSService.java:223]: Sorry, I don't understand your question.

[the remark this answers is not in the document]
Comment (jlaur) [reply bundles/org.openhab.voice.voicerss/src/main/java/org/openhab/voice/voicerss/internal/VoiceRSSTTSService.java:223]: You added `throws IllegalStateException` to the method signature. I'm just wondering about best practice here - if methods not throwing a particular exception type by themselves need to declare it when it's an unchecked exception? It doesn't do any harm, so I can approve PR with no problems, just curious about this.

[the remark this answers is not in the document]
Comment (jlaur) [reply bundles/org.openhab.voice.voicerss/src/main/java/org/openhab/voice/voicerss/internal/VoiceRSSTTSService.java:223]: https://stackoverflow.com/questions/25743574/java-best-practice-for-declaring-unchecked-exception

[the remark this answers is not in the document]
Comment (lolodomo) [reply bundles/org.openhab.voice.voicerss/src/main/java/org/openhab/voice/voicerss/internal/VoiceRSSTTSService.java:223]: Ok, I understand now.
IMHO, this is better to mention it. If not, the caller will not know that such exception can be thrown by the call.

**Review by jlaur — APPROVED** [review]
LGTM

**Comment by lolodomo** [comment]
Thank you

**Pull request merged**